tests/keras/engine/test_training.py

```diff
@@ -27,6 +32,39 @@ def __getitem__(self, idx):
             np.random.random((self.batch_size, 3))]
 
 
+def call_model_methods(model, input_np, output_np, batch_size=10, epochs=1):
+
+    # train_on_batch
+    out = model.train_on_batch(input_np,
+                               output_np)
+    out = model.train_on_batch(input_np,
+                               output_np)
+
+    # test_on_batch
+    out = model.test_on_batch(input_np,
+                              output_np)
+
+    # predict_on_batch
+    out = model.predict_on_batch(input_np)
+
+    # fit
+    out = model.fit(input_np,
+                    output_np, epochs=1, batch_size=batch_size)
+    out = model.fit(input_np,
+                    output_np, epochs=1, batch_size=batch_size)
+
+    # evaluate
+    out = model.evaluate(input_np,
+                         output_np, batch_size=batch_size)
+    out = model.evaluate(input_np,
+                         output_np, batch_size=batch_size)
+
+    # predict
+    out = model.predict(input_np, batch_size=batch_size)
```

Comment:
Also I bet predict twice ensures the first call doesn't cause something to break that won't show up until the second call.